File: MATLAB Test Scripts/atan2d_0_360_test.m
% atan2d_0_360_test
%{
    This script tests the atan2d_0_360 function for various inputs.

    t - angle in degrees

    User M-function required: atan2d_0_360
%}
% ----------------------------------------------

%...Inputs
inputs = [ 0, 0; 1, 0;   0, 1; -1, 0;
          0, -1; 1, 1; -1, -1; 1, -1;
          -1, 1; sqrt(3), 1; -sqrt(3), -1;
          sqrt(3), -1; -sqrt(3), 1];

for i = 1:size(inputs, 1)
    y = inputs(i, 1);
    x = inputs(i, 2);
    result = atan2d_0_360(y, x);

    fprintf('Test case %d: atan2d_0_360(%f, %f)\n', i, y, x);
end

File: tests/cpp/Sources/atan2d_0_360_test.cpp
#include <iostream>
#include <vector>
#include "atan2d_0_360.h"

/*
    This script tests the atan2d_0_360 function for various inputs.

    t - angle in degrees

    User h-function required: atan2d_0_360
*/
int main() 
{
    //...Inputs
    std::vector<std::pair<double, double>> inputs = 
    {
        {0, 0}, {1, 0}, {0, 1}, {-1, 0}, {0, -1},
        {1, 1}, {-1, -1}, {1, -1}, {-1, 1},
        {std::sqrt(3), 1}, {-std::sqrt(3), -1},
        {std::sqrt(3), -1}, {-std::sqrt(3), 1}
    };

    int case_num = 1;
    for (const auto& [y, x] : inputs) 
    {
        double result = atan2d_0_360(y, x);
        std::cout << "atan2d_0_360(" << y << ", " << x << ") = " << result << "\n";
    }

    return 0;
}
